cmd/progress-demo/main.go

```diff
@@ -0,0 +1,133 @@
+package main
```

Comment:
Looks to me like this is a test for the progress feature, is there a way to convert this to a test in the progress package.